src/modules/DOM.js

```diff
@@ -16,16 +14,40 @@ document.getElementById('myonoffswitch').addEventListener('change',(event)=> {
 });
 
 document.getElementById('search-ic').addEventListener('click', (event)=> {
-
+    setTimeout(toggleLoadingModal,2000);
     const searchBar = document.getElementById('search-bar');
     searchWeatherByCity(searchBar.value.trim());
```

Comment:
searchWeatherByCity and its implementation needs to be async. do this by using async/await or promises.